provider_shopper/test/catalog_widget_test.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter/material.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/catalog.dart';
+
+Widget createCatalogScreen() => MultiProvider(
+      providers: [
+        Provider(create: (context) => CatalogModel()),
+        ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            cart.catalog = catalog;
+            return cart;
+          },
+        ),
+      ],
+      child: MaterialApp(
+        home: MyCatalog(),
+      ),
+    );
+
+void main() {
+  final catalogListItems = CatalogModel.itemNames;
+
+  group('CatalogScreen Widget Tests', () {
+    testWidgets('Testing the layout and presentation of widgets',
+        (tester) async {
+      await tester.pumpWidget(createCatalogScreen());
+
+      // Verifying the layout
```

Comment:
These two expect statements aren't all that interesting (it would almost be a failure of the Flutter SDK rather than the app if they failed). I would recommend cutting them and focusing the test on making sure the catalog screen displays the right number of rows, which you're doing below.